WordPress/src/main/java/org/wordpress/android/ui/blaze/blazecampaigns/campaignlisting/CampaignListingViewModel.kt

```diff
@@ -1,123 +1,210 @@
 ) : ScopedViewModel(bgDispatcher) {
+    private lateinit var site: SiteModel
+
     private val _uiState = MutableLiveData<CampaignListingUiState>()
     val uiState: LiveData<CampaignListingUiState> = _uiState
 
     private val _navigation = MutableLiveData<Event<CampaignListingNavigation>>()
     val navigation = _navigation
 
+    private val _refresh = MutableLiveData<Boolean>()
+    val refresh = _refresh
+
+    private val _snackbar = MutableSharedFlow<String>()
+    val snackBar = _snackbar.asSharedFlow()
+
+    private var page = 1
+    private var limitPerPage: Int = 10
+    private var isLastPage: Boolean = false
+
     fun start(campaignListingPageSource: CampaignListingPageSource) {
+        this.site = selectedSiteRepository.getSelectedSite()!!
         blazeFeatureUtils.trackCampaignListingPageShown(campaignListingPageSource)
-        _uiState.postValue(CampaignListingUiState.Loading)
         loadCampaigns()
     }
 
     private fun loadCampaigns() {
-        if (!networkUtilsWrapper.isNetworkAvailable()) {
-            // showNoInternet() error, skipping for now so that loading state can be design reviewed
-            return
-        }
+        _uiState.postValue(CampaignListingUiState.Loading)
         launch {
-            val blazeCampaignModel = blazeCampaignsStore.getBlazeCampaigns(selectedSiteRepository.getSelectedSite()!!)
-            if (blazeCampaignModel.campaigns.isEmpty()) {
-                showNoCampaigns()
-            } else {
-                val campaigns = blazeCampaignModel.campaigns.map {
-                    it.mapToCampaignModel()
-                }
-                showCampaigns(campaigns)
+            when (val campaigns = getCampaignListFromDbUseCase.execute(site)) {
+                is Either.Right -> showCampaigns(campaigns.value)
+                is Either.Left -> fetchCampaigns()
             }
         }
     }
 
-    private fun BlazeCampaignModel.mapToCampaignModel(): CampaignModel {
-        return CampaignModel(
-            id = this.campaignId.toString(),
-            title = UiString.UiStringText(title),
-            status = CampaignStatus.fromString(uiStatus),
-            featureImageUrl = imageUrl,
-            impressions = mapToStatsStringIfNeeded(impressions),
-            clicks = mapToStatsStringIfNeeded(clicks),
-            budget = convertToDollars(budgetCents)
-        )
-    }
-
-    private fun mapToStatsStringIfNeeded(value: Long): UiString? {
-        return if (value != 0L) {
-            val formattedString = statsUtils.toFormattedString(value, ONE_THOUSAND)
-            UiString.UiStringText(formattedString)
+    private suspend fun fetchCampaigns() {
+        if (networkUtilsWrapper.isNetworkAvailable().not()) {
+            showNoNetworkError()
         } else {
-            null
+            when (val campaignResult = fetchCampaignListUseCase.execute(site, page)) {
+                is Either.Right -> showCampaigns(campaignResult.value)
```

Comment:
@AjeshRPai 

❓ I like the use or `Either`, but when my brain looks at this `Right` or `Left` provides no context. What are your thoughts on renaming `Left` and `Right` to `Success` and `Failure`? This can make the code more readable and better convey the meaning of the two possible outcomes.